src/server/auth/arboristClient.js

```diff
@@ -65,6 +65,64 @@ class ArboristClient {
       },
     );
   }
+
+  listAuthorizedCreateResources(jwt) {
```

Comment:
Most of this function is duplicating `listAuthorizedResources(jwt)`, except for the part that check's the user for certain service + methods.
We don't need to duplicate those codes, we should be extracting the part of making request to arborist and checking for user's access in the returned auth mapping into two separated functions, and just reuse them. In this way, we can avoid sending duplicated requests to arborist
See how the fetch for user's auth mapping and check for user's access can be separated in another service
https://github.com/uc-cdis/data-portal/blob/master/src/actions.js#L513
https://github.com/uc-cdis/data-portal/blob/master/src/authMappingUtils.js#L53

Reply:
Ok getAuthMapping function has been created and used twice, once for the reader permissions checking, and again for the guppy admin permissions checking. See https://github.com/uc-cdis/guppy/pull/273/commits/713fc1fac7a91b471b6c8696843e7c9d23a407d9